Common/Tests/Utilities/Mocks/MockServiceProvider.cs

```diff
@@ -31,32 +32,45 @@ public class MockServiceProvider : IServiceProvider, IServiceContainer {
         public object GetService(Type serviceType) {
             object service;
             Console.WriteLine("MockServiceProvider.GetService({0})", serviceType.Name);
-            Services.TryGetValue(serviceType.GUID, out service);
-            return service;
+            if (Services.TryGetValue(serviceType.GUID, out service)) {
```

Comment:
do you need to handle concurrent access to GetService()?

Reply:
Haven't needed to so far, and no evidence that we need it.

Reply:
ok